plugin/src/main/java/org/opensearch/ml/sdkclient/XContentClient.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.sdkclient;
+
+import static org.opensearch.action.DocWriteResponse.Result.CREATED;
+import static org.opensearch.action.DocWriteResponse.Result.DELETED;
+import static org.opensearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE;
+import static org.opensearch.common.xcontent.json.JsonXContent.jsonXContent;
+import static org.opensearch.core.xcontent.ToXContent.EMPTY_PARAMS;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.opensearch.action.delete.DeleteRequest;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.client.Client;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.sdk.DeleteCustomRequest;
+import org.opensearch.sdk.DeleteCustomResponse;
+import org.opensearch.sdk.GetCustomRequest;
+import org.opensearch.sdk.GetCustomResponse;
+import org.opensearch.sdk.PutCustomRequest;
+import org.opensearch.sdk.PutCustomResponse;
+import org.opensearch.sdk.SdkClient;
+
+public class XContentClient extends SdkClient {
```

Comment:
No, because `NodeClient` is an implementation. This is a wrapper around `o.o.client.Client` interface.  The implementation is not defined.  (Yeah, there's an `AbstractClient` subclass and `NodeClient` is the only core subclass, but this code doesn't care about that.).  `OpenSearchClient` has specific meaning related to the java client so is not a good choice (in fact, that's probably what I will name the remote cluster http implementation).

This client manipulates the `_source` field containing structured object data in one of four mediatypes, defined by a content-type field, often JSON but possibly YAML or others.  Perhaps `StructuredObjectClient` or `MediaTypeClient` might be better?

I'm happy to entertain naming conversations but don't want to bikeshed too much here.

Reply:
Client implementation should provide information about the data store it supports. XContent client provides information that it operates on the _source field but lacks information about the underlying data store.

Reply:
> the underlying data store.

How about `ClusterIndexClient`?

Reply:
Sounds good, thanks.